### PR TITLE
Sstack

### PR DESCRIPTION
**Pull request by FoleyLab**
## Description
This pull requests adds functionality to enable design of selective mirrors, potentially for polaritonic chemistry applications

## Todos (checkmarks indicate that unit tests are provided for these, open boxes indicate unit tests are not provided yet)
  - [x] Added method compute_selective_mirror_fom() to em.py; test_compute_selective_mirror_fom() in test_em.py
          - computes reflection efficiency using user-specified window of high reflectivity
          - compute transmission efficiency using user-specified window of high transmissivity
          - computes composite figure of merit as the weighted average of the the reflection and transmission efficiencies
          - allows users to specify custom weights for each figure of merit
  - [x]  Added method compute_selective_mirror_fom_gradient to em.py;  test_compute_selective_mirror_fom_gradient() in test_em.py
         - computes the analytic gradient of the reflection efficiency
         - computes the analytic gradient of the transmission efficiency
         - computes the analytic gradient of the selective mirror fom
  - [ ] Added lines to parse_input to collect user information about spectral regions where high transmissivity is desired and where high reflectivity is desired, stored in .transmissive_envelope and .reflective_envelope, respectively
  - [X] Added several materials of interest for cavities; unit tests added to test_materials.py
          - Si3N4 
          - ZrO2
          - SiO2 with a universal dispersion model (SiO2_UDM)
          - Al2O3 with a universal dispersion model (Al2O3_UDM)

## Questions
- [ ] Is a unit test for the specification of the reflectivity and transmissivity envelope needed?

## Status
- [ ] Ready to go

### CODE REVIEW

File: examples/Cooling_Demo.ipynb
Comment (AnonymousOceans) [inline examples/Cooling_Demo.ipynb:1]: Contains Data- Line 67

**Review by AnonymousOceans — CHANGES_REQUESTED** [review]

**Review by AnonymousOceans — APPROVED** [review]

File: examples/mirror_basin_hopping.py
Comment (AnonymousOceans) [inline examples/mirror_basin_hopping.py:1]: Maybe exclude this file as it does not demonstrate a particular feature.

File: wptherml/data/material_references.md
Comment (AnonymousOceans) [inline wptherml/data/material_references.md:1]: Add universal dispersion models for glass and aluminum oxide.

File: wptherml/em.py
Comment (AnonymousOceans) [inline wptherml/em.py:1]: Add equations to WPTherml Equation Sheet.

**Review by AnonymousOceans — CHANGES_REQUESTED** [review]
Add missing file information.

**Review by rubymanderna — APPROVED** [review]
Everything looks good

**Comment by rubymanderna** [comment]
I approve, ready to go.

**Review by AnonymousOceans — APPROVED** [review]
Good to go!

**Comment by AnonymousOceans** [comment]
Good to go!

**Review by FoleyLab — COMMENTED** [review]
GTG

**Pull request merged**